src/m365/todo/commands/task/task-add.ts

```diff
@@ -0,0 +1,126 @@
+import * as chalk from 'chalk';
+import { Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  title: string;
+  listName?: string;
+  listId?: string;
+}
+
+class TodoTaskAddCommand extends GraphCommand {
+  public get name(): string {
+    return `${commands.TASK_ADD}`;
+  }
+
+  public get description(): string {
+    return 'Add a task to a Microsoft To Do task list';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.listId = typeof args.options.listId !== 'undefined';
+    telemetryProps.listName = typeof args.options.listName !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    const endpoint: string = `${this.resource}/v1.0`;
+
+    let id: string | undefined;
+
+    const data: any = {
+      title: args.options.title
+    };
+
+    ((): Promise<any> => {
+      if (args.options.listName) {
+        const requestOptions: any = {
+          url: `${endpoint}/me/todo/lists?$filter=displayName eq '${args.options.listName}'`,
```

Comment:
Since the list name has been provided by the user, we should escape it before passing on to the query